src/cogent3/core/location.py

```diff
@@ -1596,6 +1591,202 @@ def make_seq_feature_map(self, align_feature_map: "FeatureMap") -> "FeatureMap":
+            )
+        if self.num_gaps == 0 or other.num_gaps == 0:
+            return numpy.array([], dtype=_DEFAULT_GAP_DTYPE)
+
+        return self.shared_gaps(other.get_gap_align_coordinates())
+
+    @shared_gaps.register
+    def _(self, other: numpy.ndarray) -> numpy.ndarray:
+        """returns a numpy array of the shared [(gap start, gap end), ...]
+
+        Notes
+        -----
+        The result is in alignment coordinates
+        """
+        other_gaps = other
+        if not len(other_gaps):
+            return numpy.array([], dtype=_DEFAULT_GAP_DTYPE)
+
+        if other_gaps[-1][-1] > len(self):
+            raise AssertionError(
+                f"other_gaps {other_gaps!r} from a different alignment?"
+            )
+        self_gaps = self.get_gap_align_coordinates()
+
+        result = coords_intersect(self_gaps, other_gaps)
+        return numpy.array(result, dtype=_DEFAULT_GAP_DTYPE)
+
+    @functools.singledispatchmethod
+    def minus_gaps(self, other_gaps):
+        """returns new map with gaps in other_gaps removed from self
+
+        Parameters
+        ----------
+        other_gaps
+            IndelMap instance, or numpy array in alignment coordinates
+
+        Returns
+        -------
+        New instance with gap spans represented by other_gaps removed from self
+        """
+        if len(self) != len(other_gaps):
+            raise AssertionError(
+                f"{len(other_gaps)=} != {len(self)=}, from a different alignment?"
+            )
+
+        return self.minus_gaps(other_gaps.get_gap_align_coordinates())
+
+    @minus_gaps.register
+    def _(self, other_gaps: numpy.ndarray):
+        if not len(other_gaps):
+            return self
+
+        if other_gaps[-1][-1] > len(self):
+            raise AssertionError(
+                f"other_gaps {other_gaps!r} from a different alignment?"
+            )
+        self_gaps = self.get_gap_align_coordinates()
+        unique = coords_minus_coords(self_gaps, other_gaps)
+        new_gaps = numpy.empty((len(unique), 2), dtype=_DEFAULT_GAP_DTYPE)
+        for i, (start, end) in enumerate(unique):
+            new_gaps[i] = self.get_seq_index(start), end - start
+        gap_pos, gap_lengths = new_gaps.T
+        return self.__class__(
+            gap_pos=gap_pos,
+            gap_lengths=gap_lengths,
+            parent_length=self.parent_length,
+        )
+
+
+_empty = None, None
+
+
+def coords_minus_coords(
+    coords1: numpy.ndarray, coords2: numpy.ndarray
+) -> numpy.ndarray:
+    """returns the coords1 minus any overlaps with coords2
+
+    Parameters
+    ----------
+    coords1, coords2
+        series of coordinates defining segments
+
+    Notes
+    ------
+    Assumes both coordinate series are sorted.
+    """
+    dtype = getattr(coords1, "dtype", _DEFAULT_GAP_DTYPE)
+    unique_segments = []
+    for a1, a2 in coords1:
+        total_intersect = None
+        for b1, b2 in coords2:
+            if b2 < a1:
```

Comment:
not sure why here is strictly less than, where below it is less than or equal to

Reply:
a1 and b1 are "starts", a2 and b2 are "ends. This condition is the end of segment *b* is before the start of segment *a*. Because a2 is not inclusive, the condition below is saying segment *a* and *b* do not intersect